File: iOS链式编程/CalculatorForOC/CalculatorForOC/NSObject+XZCalculator.m
//
//  NSObject+XZCalculator.m
//  CalculatorForOC
//
//  Created by aa on 2017/11/7.
//  Copyright © 2017年 aa. All rights reserved.
//

#import "NSObject+XZCalculator.h"

@implementation NSObject (XZCalculator)

- (void)xz_makeCalculation:(void(^)(XZCalculator *calculator))block{
    XZCalculator *calculator = [[XZCalculator alloc] init];
    block(calculator);
}
@end

File: iOS链式编程/CalculatorForSwift/CalculatorForSwift/NSObject+XZCalculator.swift
//
//  Any+XZCalculator.swift
//  CalculatorForSwift
//
//  Created by aa on 2017/11/7.
//  Copyright © 2017年 aa. All rights reserved.
//

import Foundation

extension NSObject {
    func xz_makeCalculation(_ block:(_ calculator:XZCalculator) -> ()){
        let calculator:XZCalculator = XZCalculator();
        block(calculator);
    }
}
